x-pack/plugins/security_solution/public/cloud_security_posture/components/csp_details/misconfiguration_findings_details_table.tsx

```diff
@@ -0,0 +1,189 @@
+import { CspEvaluationBadge } from '@kbn/cloud-security-posture';
+
+const getFindingsStats = (passedFindingsStats: number, failedFindingsStats: number) => {
+  if (passedFindingsStats === 0 && failedFindingsStats === 0) return [];
+  return [
+    {
+      key: i18n.translate(
+        'xpack.securitySolution.flyout.right.insights.misconfigurations.passedFindingsText',
+        {
+          defaultMessage: 'Passed findings',
+        }
+      ),
+      count: passedFindingsStats,
+      color: euiThemeVars.euiColorSuccess,
+    },
+    {
+      key: i18n.translate(
+        'xpack.securitySolution.flyout.right.insights.misconfigurations.failedFindingsText',
+        {
+          defaultMessage: 'Failed findings',
+        }
+      ),
+      count: failedFindingsStats,
+      color: euiThemeVars.euiColorVis9,
+    },
+  ];
+};
+
+/**
+ * Insights view displayed in the document details expandable flyout left section
+ */
+export const MisconfigurationFindingsDetailsTable = memo(
+  ({ fieldName, queryName }: { fieldName: 'host.name' | 'user.name'; queryName: string }) => {
+    const { data } = useMisconfigurationFindings({
+      query: buildEntityFlyoutPreviewQuery(fieldName, queryName),
+      sort: [],
+      enabled: true,
+      pageSize: 1,
+    });
+
+    const passedFindings = data?.count.passed || 0;
+    const failedFindings = data?.count.failed || 0;
+
+    const [pageIndex, setPageIndex] = useState(0);
+    const [pageSize, setPageSize] = useState(10);
+
+    const findingsPagination = (
+      findings: Array<Pick<CspFinding, 'result' | 'rule' | 'resource'>>
+    ) => {
+      let pageOfItems;
+
+      if (!pageIndex && !pageSize) {
+        pageOfItems = findings;
+      } else {
+        const startIndex = pageIndex * pageSize;
+        pageOfItems = findings?.slice(
+          startIndex,
+          Math.min(startIndex + pageSize, findings?.length)
+        );
+      }
+
+      return {
+        pageOfItems,
+        totalItemCount: findings?.length,
+      };
+    };
+
+    const { pageOfItems, totalItemCount } = findingsPagination(data?.rows || []);
+
+    const pagination = {
+      pageIndex,
+      pageSize,
+      totalItemCount,
+      pageSizeOptions: [10, 25, 100],
+    };
+
+    const onTableChange = ({
+      page,
+    }: Criteria<Pick<CspFinding, 'result' | 'rule' | 'resource'>>) => {
+      if (page) {
+        const { index, size } = page;
+        setPageIndex(index);
+        setPageSize(size);
+      }
+    };
+
+    const navToFindings = useNavigateFindings();
+
+    const navToFindingsByHostName = (hostName: string) => {
+      navToFindings({ 'host.name': hostName }, ['rule.name']);
+    };
+
+    const navToFindingsByRuleAndResourceId = (ruleId: string, resourceId: string) => {
+      navToFindings({ 'rule.id': ruleId, 'resource.id': resourceId });
+    };
+
+    const columns: Array<EuiBasicTableColumn<Pick<CspFinding, 'result' | 'rule' | 'resource'>>> = [
```

Comment:
this type `Pick<CspFinding, 'result' | 'rule' | 'resource'>` is being reused multiple times, what about create one type for it: 

```ts
type MisconfigurationFindingDetailFields = Pick<CspFinding, 'result' | 'rule' | 'resource'>
```